spring-vault-core/src/main/java/org/springframework/vault/core/lease/domain/RequestedSecret.java

```diff
@@ -66,6 +66,20 @@ public static RequestedSecret rotating(String path) {
 	}
 
 	/**
+	 * Create a rotating or renewable {@link RequestedSecret} at {@code path}. A lease associated with
+	 * this secret will be renewed if the lease is qualified for renewal. Once the lease
+	 * expires, a new secret with a new lease is obtained if mode is ROTATE, otherwize the lease is no
+	 * longer valid after expiry.
+	 *
+	 * @param mode must not be {@literal null}
+	 * @param path must not be {@literal null} or empty, must not start with a slash.
+	 * @return the rotating {@link RequestedSecret}.
+	 */
+	public static RequestedSecret buildFromMode(Mode mode, String path) {
```

Comment:
How about just `from(Mode, Path)`?

I'd propose to use a simple `if` construction instead of introducing API methods on `Mode`. You might want to `Assert.notNull(mode, …)` here as well.